05-ts/30-UiSelect/TestComponent.vue

```diff
@@ -0,0 +1,56 @@
+<script setup lang="ts">
+import { reactive, ref, toRefs } from "vue";
+import type { Ref } from 'vue';
+
+type SelectValue = null | number;
+type SingleOption = {
+    label: string,
+    value: number
+};
+
+const tempTest = reactive({
+    select1: null as SelectValue,
+    select2: null as SelectValue,
+    select3: null as SelectValue,
+    select4: null as SelectValue,
+})
+
+const {select1, select2, select3, select4} = toRefs(tempTest);
```

Comment:
Переменная `tempTest` здесь не нужна, она никак не используется, кроме как для деструктуризации позже.

Тип лучше определять более явно, без `as`. С `as` будет пропущена ошибка, если `null` несовместим с типом `SelectValue`.

Например:

```suggestion
const select1 = ref<SelectValue>(null)
const select2 = ref<SelectValue>(null)
const select3 = ref<SelectValue>(null)
const select4 = ref<SelectValue>(null)
```